numpy/core/tests/test_regression.py

```diff
@@ -1408,6 +1408,11 @@ def test_unicode_to_string_cast(self):
                      dtype='U')
         assert_raises(UnicodeEncodeError, np.array, a, 'S4')
 
+    def test_unicode_to_string_cast_error(self):
+        # Ticket #15790.
```

Comment:
```suggestion
        # gh-15790.
```

"Ticket" refers to an old non-github issue tracker.

Reply:
Should I update the commit messages too? They use "#15790".

Reply:
I guess we/I like `gh-` but its also a bit of a lost cause :). So if you are motivated, sure. The word "Ticket" explicitly feels like the very old Ticket system...

Reply:
Indeed, the real danger is the word `Ticket`. #NNN vs gh-NNN isn't quite so big an issue, but the former is a little more future-proof, and looks better than `# #15790` in a comment.

Reply:
I probably won't update the commit messages just because I don't like changing history in the middle of a PR.

I'll update this, though.

Reply:
If you're concerned about a nice history, it might be neat to rewrite these commits - start wiht the `MAINT` commit first which eliminates a bunch of code, and then the `BUG`/`TST` second.

Reply:
Sorry, are you saying I *should* rewrite/reorder the commits? I can if that would help, but I just want to clarify first.

Reply:
If you don't, I'll probably just squash them into a single commit, and I'm fine with doing that.

In the context of a merged PR, I find it's good for commits to tell a story of the shortest path from start to finish, not the longer path that was taken in response to review feedback. Adding a bunch of code only to simplify it and remove it again just creates noise for people walking back through the blame.

Reply:
Gotcha. I'll take a second and do that then, since this is a pretty simple patch.

I just know that other projects don't like force-pushed changes (since it can mess up the review), so I wanted to check first!

Reply:
I’ve completed the rebase. Everything looks good!

Reply:
One last request - can you move the `TST` commit after the `MAINT` commit so that the `MAINT` commit would pass by itself?

Reply:
Hmm... It looks like that's just a display issue in the PR. If you go to my branch (or check it out locally), you can see that the commits are in the [right order](https://github.com/brandtbucher/numpy/commits/unsafe-cast).

I've seen similar things happen on GitHub with complicated rebases and force-pushes before.

Reply:
Haven't seen that one in a while, thanks for the reminder!